src/Servers/Kestrel/Core/src/Internal/KestrelServerImpl.cs

```diff
@@ -219,7 +222,7 @@ async Task OnBind(ListenOptions options, CancellationToken onBindCancellationTok
                         // Add the connection limit middleware
                         connectionDelegate = EnforceConnectionLimit(connectionDelegate, Options.Limits.MaxConcurrentConnections, Trace);
 
-                        options.EndPoint = await _transportManager.BindAsync(options.EndPoint, connectionDelegate, options.EndpointConfig, onBindCancellationToken).ConfigureAwait(false);
+                        options.EndPoint = await _transportManager.BindAsync(ResolveEndPoint(originalEndPoint, options.EndPoint), connectionDelegate, options.EndpointConfig, onBindCancellationToken).ConfigureAwait(false);
```

Comment:
I don't think we need to call `ResolveEndPoint` here since `options.EndPoint` hasn't changed yet. I agree with @Tratcher's comment about filing a bug to fix `MsQuicAddressHelpers`. Once that's filed, let's add a comment linking to it.